frontend/src/metabase/visualizations/components/legend/LegendLayout.jsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import PropTypes from "prop-types";
+import _ from "underscore";
+import ExplicitSize from "metabase/components/ExplicitSize";
+import Legend from "./Legend";
+import {
+  ChartContainer,
+  LegendContainer,
+  LegendLayoutRoot,
+} from "./LegendLayout.styled";
+
+const MIN_ITEM_WIDTH = 100;
+
+const propTypes = {
+  className: PropTypes.string,
+  labels: PropTypes.array.isRequired,
+  colors: PropTypes.array.isRequired,
+  hovered: PropTypes.object,
+  width: PropTypes.number,
+  hasLegend: PropTypes.bool,
+  actionButtons: PropTypes.node,
+  children: PropTypes.node,
+  onHoverChange: PropTypes.func,
+  onAddSeries: PropTypes.func,
+  onSelectSeries: PropTypes.func,
+  onRemoveSeries: PropTypes.func,
+};
+
+const LegendLayout = ({
+  className,
+  labels,
+  colors,
+  hovered,
+  width,
+  hasLegend,
+  actionButtons,
+  children,
+  onHoverChange,
+  onAddSeries,
+  onSelectSeries,
+  onRemoveSeries,
+}) => {
+  const isNarrow = labels.length * MIN_ITEM_WIDTH > width;
+  const isVertical = false;
```

Comment:
In the future `isNarrow` will be removed and `isVertical` will be calculated correctly. This is just to give the full picture of upcoming changes.